packages/did-auth-siop-op-authenticator/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -0,0 +1,204 @@
+        response.status >= 400 ? Promise.reject(new Error(await response.text())) : this.op.parseAuthenticationRequestURI(await response.text())
+      )
+      .catch((error: unknown) => Promise.reject(error))
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.getDidSiopAuthenticationRequestDetails} */
+  private async getDidSiopAuthenticationRequestDetails(
+    args: getDidSiopAuthenticationRequestDetailsArgs,
+    context: IRequiredContext
+  ): Promise<IAuthRequestDetails> {
+    const presentationDefs = args.verifiedAuthenticationRequest.presentationDefinitions
+    const verifiablePresentations =
+      presentationDefs && presentationDefs.length > 0 ? await this.matchPresentationDefinitions(presentationDefs, args.verifiableCredentials) : []
+    const didResolutionResult = args.verifiedAuthenticationRequest.didResolutionResult
+
+    return {
+      id: didResolutionResult.didDocument!.id,
+      alsoKnownAs: didResolutionResult.didDocument!.alsoKnownAs,
+      vpResponseOpts: verifiablePresentations,
+    }
+  }
+
+  private async matchPresentationDefinitions(
+    presentationDefs: PresentationDefinitionWithLocation[],
+    verifiableCredentials: VerifiableCredential[]
+  ): Promise<IMatchedPresentationDefinition[]> {
+    const presentationExchange = this.getPresentationExchange(verifiableCredentials)
+    return await Promise.all(
+      presentationDefs.map(async (presentationDef: PresentationDefinitionWithLocation) => {
+        const checked = await presentationExchange.selectVerifiableCredentialsForSubmission(presentationDef.definition)
+        if (checked.errors && checked.errors.length > 0) {
+          return Promise.reject(new Error(JSON.stringify(checked.errors)))
+        }
+
+        const matches: SubmissionRequirementMatch[] | undefined = checked.matches
+        if (matches && matches.length == 0) {
+          return Promise.reject(new Error(JSON.stringify(checked.errors)))
+        }
+
+        const verifiablePresentation = await presentationExchange.submissionFrom(presentationDef.definition, verifiableCredentials)
+        return {
+          location: presentationDef.location,
+          format: VerifiablePresentationTypeFormat.LDP_VP,
+          presentation: verifiablePresentation,
+        }
+      })
+    )
+  }
+
+  private getPresentationExchange(verifiableCredentials: VerifiableCredential[]): PresentationExchange {
+    return new PresentationExchange({
+      did: this.op.authResponseOpts.did,
+      allVerifiableCredentials: verifiableCredentials,
+    })
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.verifyDidSiopAuthenticationRequestURI} */
+  private async verifyDidSiopAuthenticationRequestURI(
+    args: IVerifyDidSiopAuthenticationRequestUriArgs,
+    context: IRequiredContext
+  ): Promise<VerifiedAuthenticationRequestWithJWT> {
+    const didMethodsSupported = args.requestURI.registration?.did_methods_supported as string[]
+    let didMethods: string[] = []
```

Comment:
Hmzz, what we are doing here is the RP is stating its supported DIDs. We are now adding that to our Veramo/RP list and resolver. What we should be doing is verifying it against locally (RP/Veramo) supported DIDs first and only choose those that we actually support. As long as we use our Universal DID client in the Agent we should be good for now, but it is wise to add a fixme and/or ticket here

Reply:
Ideally what should happen here is that we expose which identifiers are potential matches to the did_methods_supported.
The RP lists which DID methods are supported. The DIDManager knows which DIDs it has. So it could filter the list to only the DIDs that have a DID method from the did_methods_supported. If dids_methods_supported is not provided/empty it means all DIDs could be used

Reply:
Added a todo with a ticket link